### ISSUE TITLE
addr2line not found (failing to start EU IV in Wayland session)

### ISSUE THREAD

**Issue by FelixSchwarz** [flathub/com.valvesoftware.Steam#342]
**Game information**: Europa Universalis IV
**Distribution name and version where applicable**: Fedora 29 (x86_64)

**Flatpak info**: 

    $ flatpak --version
    Flatpak 1.2.4
    $ flatpak --gl-drivers
    default
    host

**Problem description**
Europa Universalis IV and other games are failing to start inside a Wayland session (see https://bugs.freedesktop.org/show_bug.cgi?id=98272).

Besides that problem I see that Steam(?) tries to run "addr2line" while starting the game:

     sh: addr2line: command not found

Please note: The problem is only visible when run with a Wayland session. This does not happen with Xorg.

Full log: [steam.log](https://github.com/flathub/com.valvesoftware.Steam/files/3048190/steam.log)

**Does this issue reproduce with native Steam**: have not tested

**Comment by nanonyme** [comment]
This?

**Comment by FelixSchwarz** [comment]
yes, I guess. I only see the line I quoted but I guess it has something to do with the game failure. I suspect Steam tries to gather some crash/debug data (via addr2line).

**Comment by nanonyme** [comment]
Could you please test the related PR? You probably need to uninstall the Steam app to test.

**Comment by nanonyme** [comment]
Even if it doesn't fix the problem but gives better log output, we could potentially keep it.

**Comment by FelixSchwarz** [comment]
> Could you please test the related PR? You probably need to uninstall the Steam app to test.

I tried but after installing the updated flatpak, I only see this error and Steam won't start at all
(full logs: [error.txt](https://github.com/flathub/com.valvesoftware.Steam/files/3049814/error.txt)):

    You are missing the following 32-bit libraries, and Steam may not run:
    libc.so.6

**Comment by nanonyme** [comment]
That's a fake error based on my reading up on .local/share/Steam/steam.sh. You can find this inside .var/app com.valvesoftware.Steam. It would be great if you could remove output suppression from the line that calls ldd on steam. It's possible compat extension isn't working (is it installed?) as expected when you install using those guides.

**Comment by nanonyme** [comment]
Maybe I should sanity check runtime extensions myself and not let Steam do that.

**Comment by FelixSchwarz** [comment]
> It would be great if you could remove output suppression from the line that calls ldd on steam.

I just wanted to do that but before I ran "flatpak update" and tried to run application again (to establish a baseline). Et voilà, the "ldd" error messages were gone (as well as Steam's complaint about missing 32-bit libraries).

So +1 from me for pull request #343.

**Comment by nanonyme** [comment]
So the game now works, right?

**Comment by FelixSchwarz** [comment]
No - this will only change when https://bugs.freedesktop.org/show_bug.cgi?id=98272 is fixed. However now the Steam flatpak shows the same behavior as the native version (no addr2line errors).

**Comment by nanonyme** [comment]
Is the game bundling SDL2? It seems probable to me in this case org.freedesktop.Platform SDL2 doesn't get loaded due to library load order. We're going to be adding more magic soon that might allow better than native functionality. Let's keep this open for now

**Comment by nanonyme** [comment]
As in, Steam SDL2 is built without Wayland support but we have it so we might be able to make this now use XWayland

**Comment by FelixSchwarz** [comment]
> Is the game bundling SDL2?

I'm not sure how to check that.

This is the ldd output for the main executable:

    ldd eu4
	    linux-gate.so.1 (0xf7eda000)
	    libdl.so.2 => /lib/libdl.so.2 (0xf7eab000)
	    libX11.so.6 => /lib/libX11.so.6 (0xf7d54000)
	    libpthread.so.0 => /lib/libpthread.so.0 (0xf7d32000)
	    libsteam_api.so => …/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/Europa Universalis IV/./libsteam_api.so (0xf7d09000)
	    libPDXBrowser_IPC.so => …/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/Europa Universalis IV/./libPDXBrowser_IPC.so (0xf7cd3000)
	    libGLU.so.1 => not found
	    libGL.so.1 => /lib/libGL.so.1 (0xf7c6d000)
	    libpops_api.so => …/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/Europa Universalis IV/./libpops_api.so (0xf761d000)
	    libtbb.so.2 => …/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/Europa Universalis IV/./libtbb.so.2 (0xf75de000)
	    libm.so.6 => /lib/libm.so.6 (0xf750b000)
	    libgcc_s.so.1 => /lib/libgcc_s.so.1 (0xf74ed000)
	    libc.so.6 => /lib/libc.so.6 (0xf7343000)
	    /lib/ld-linux.so.2 (0xf7edb000)
	    libxcb.so.1 => /lib/libxcb.so.1 (0xf7314000)
	    libstdc++.so.6 => /lib/libstdc++.so.6 (0xf7180000)
	    librt.so.1 => /lib/librt.so.1 (0xf7174000)
	    libuuid.so.1 => /lib/libuuid.so.1 (0xf716a000)
	    libGLX.so.0 => /lib/libGLX.so.0 (0xf7146000)
	    libXext.so.6 => /lib/libXext.so.6 (0xf7130000)
	    libGLdispatch.so.0 => /lib/libGLdispatch.so.0 (0xf70ce000)
	    libz.so.1 => /lib/libz.so.1 (0xf70b2000)
	    libresolv.so.2 => /lib/libresolv.so.2 (0xf7098000)
	    libXau.so.6 => /lib/libXau.so.6 (0xf7092000)

"find" does not find any file with "SDL" or "sdl" in the game directory.

**Comment by nanonyme** [comment]
Ack. So we're probably talking of libSDL from Steam runtime again here. That has no Wayland support despite requests.

**Comment by nanonyme** [comment]
The root problem here is swrast gets loaded at all. That means software rendering.

**Comment by nanonyme** [comment]
https://cgit.freedesktop.org/xorg/xserver/commit/?id=8469241592b94b002a975274a596ca0dcdd9563f looks like a related fix on host side.

**Comment by nanonyme** [comment]
Does this still reproduce?

**Comment by FelixSchwarz** [comment]
Unfortunately yes. I just tried with Fedora 32 and the latest updates (Fedora, Steam flatpak, game updates). I can see the Paradox launcher but once I hit "Play" the game crashes and I see the Paradox crash reporter asking me to add additional comments about the crash.

However judging from the linked bug reports I guess this might be another bug but it still does not work. I might work with the very latest packages based on [another bug report](https://gitlab.freedesktop.org/xorg/xserver/-/issues/991). Fedora 32 comes with X Server 1.20.8. However `glxgears -samples 4` does not work for me with Wayland.
